docs/reference-guides/data/data-core-block-editor.md

```diff
@@ -738,6 +738,35 @@ _Returns_
 
 Returns the currently selected block, or null if there is no selected block.
 
+_Usage_
+
+```js
+import { select } from '@wordpress/data';
+import { store as blockEditorStore } from '@wordpress/block-editor';
+
+// Set initial active block client ID
+let activeBlockClientId = null;
+
+const getActiveBlockData = () => {
+	const activeBlock = select( blockEditorStore ).getSelectedBlock();
+
+	if ( activeBlock && activeBlock.clientId !== activeBlockClientId ) {
+		activeBlockClientId = activeBlock.clientId;
+
+		// Get active block name and attributes
+		const activeBlockName = activeBlock.name;
+		const activeBlockAttributes = activeBlock.attributes;
+
+		// Log active block name and attributes
+		console.log( activeBlockName, activeBlockAttributes );
+	}
+};
+
+wp.data.subscribe( () => {
```

Comment:
I'm not sure we need this part it doesn't add much to the example and might be taken as being needed to use it.

Reply:
Without a listener the function wouldn't log anything to the console, the subscribe would be needed here.

Reply:
Right, but this could also be assigned to a button onClick or run in a useEffect for example.

Looking at some other [examples](https://developer.wordpress.org/block-editor/reference-guides/data/data-core/#getposttypes), we don't always show how the implementation should be used - just how to implement the item in question.

That being said, this is a strong opinion loosely held so I'll leave it up to you :) If you decide to keep it can we change the `subscribe` syntax to use imports instead of accessing the global?

Great work and keep them coming!

Reply:
If we remove subscribe is there a preferred way to add a note relating to a couple of use cases. I never thought about using it in an onClick, might be good to show both options in the docs.

Thoughts?

Reply:
You could add it commented out similar to how it's [shown here](https://github.com/WordPress/wporg-developer-blog/issues/68)

Reply:
> You could add it commented out similar to how it's [shown here](https://github.com/WordPress/wporg-developer-blog/issues/68)

That the correct link? Seems unrelated

Reply:
🤦 this is the correct link https://developer.wordpress.org/block-editor/reference-guides/packages/packages-core-data/#useentityrecord

Reply:
Thanks, added a couple of examples of how the function can be used